packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -447,6 +452,12 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
     Size bottomSheetSize = Size.zero;
     Size snackBarSize = Size.zero;
 
+    // Set the size of the SnackBar early if the behavior is fixed so
+    // the FAB can be positioned based on SnackBar availability.
+    if (hasChild(_ScaffoldSlot.snackBar) && !isSnackBarFloating) {
```

Comment:
This actually is necessary, otherwise the FAB's position will move when behavior is floating. The snackbar size is taken into account when deciding where to layout the FAB

Reply:
I see, thanks for adding the comment!